src/legacy/core_plugins/metrics/index.ts

```diff
@@ -45,7 +45,7 @@ export default function(kibana: any) {
 
     init(server: Legacy.Server) {
       const initializerContext = {} as PluginInitializerContext;
-      const core = { http: { server } } as InternalCoreSetup;
+      const core = { http: { server } } as CoreSetup | any; // TODO: Remove 'any' as soon as CoreSetup is completed.
```

Comment:
Maybe `Partial<CoreSetup>` would be better as it is a little less permissive?

Reply:
@lukeelmers Every day I learn something new! Thanks for suggestion. But as I've found out the `Partial` utility couldn't skip deep properties. The issue is `http` exists in the 'CoreSetup'. Compiler complains at that. One hypothetical way is extending the `http` property. But it's not clear how to do it carefully.

Reply:
Left the comment